video/out/hwdec/hwdec_drmprime.c

```diff
@@ -167,7 +167,17 @@ static int mapper_init(struct ra_hwdec_mapper *mapper)
     struct dmabuf_interop_priv *p = mapper->priv;
 
     mapper->dst_params = mapper->src_params;
-    mapper->dst_params.imgfmt = mapper->src_params.hw_subfmt;
+
+    /*
+     * NV12 functions identically to rpi4_8 and rpi4_10. Those two pixel
```

Comment:
nit: "rpi4_8 and rpi4_10 function identically to NV12", otherwise it reads like NV12 is the weird format.

And I just want to ask for the record - I would naively expect rpi4_10 to work like P010 rather than NV12, so I'm slightly surprised that everything works for it.

Reply:
I tried setting it to P010 instead of NV12 and I couldn't tell a difference with my naked eye. I could do it this way if it's better:
```
if (strcmp(fmt_name, "rpi4_8") == 0)
    mapper->dst_params.imgfmt = IMGFMT_NV12;
else if (strcmp(fmt_name, "rpi4_10") == 0)
    mapper->dst_params.imgfmt = IMGFMT_P010;
else
    mapper->dst_params.imgfmt = mapper->src_params.hw_subfmt;
```
This would also require adding `MP_TARRAY_APPEND(p, p->formats, num_formats, IMGFMT_P010);`

Reply:
Ah yes, we don't allow it anyway. So yes, just stick with NV12. We'd need to generally validate 10bit elsewhere before that's meaningful.